src/test/java/org/takes/facets/hamcrest/HmRqBodyTest.java

```diff
@@ -67,5 +71,28 @@ public void testsBodyValuesAreDifferent() {
             Matchers.not(new HmRqBody("that"))
         );
     }
-}
 
+    /**
+     * HmRqBody can describe mismatch in descriptive way.
+     */
+    @Test
+    public void mismatchMessageIsReadable() {
+        final String was = "other";
+        final Request request = new RqFake(
+            Collections.<String>emptyList(),
+            was
+        );
+        final HmRqBody matcher = new HmRqBody("some");
+        matcher.matchesSafely(request);
+        final StringDescription description = new StringDescription();
+        matcher.describeMismatchSafely(request, description);
+        final String expected = new StringBuilder()
+            .append("body was: ")
+            .append(Arrays.toString(was.getBytes(Charset.defaultCharset())))
+            .toString();
+        MatcherAssert.assertThat(
+            description.toString(),
+            Matchers.equalTo(expected)
```

Comment:
@t-izbassar it's better to expect some constant values in test assertions instead of building expecting-result, because it will help to avoid hidden side affects if runtime changed and it's much easier to use unit-tests as a code documentation.